testing/endpoint.go

```diff
@@ -587,18 +591,68 @@ func (endpoint *Endpoint) QueryChannelUpgradeProof() ([]byte, []byte, clienttype
 
 // ChanUpgradeInit sends a MsgChannelUpgradeInit on the associated endpoint.
 // A default upgrade proposal is used with overrides from the ProposedUpgrade
-// in the channel config.
+// in the channel config, and submitted via governance proposal
 func (endpoint *Endpoint) ChanUpgradeInit() error {
 	upgrade := endpoint.GetProposedUpgrade()
 
+	// create upgrade init message via gov proposal and submit the proposal
 	msg := channeltypes.NewMsgChannelUpgradeInit(
 		endpoint.ChannelConfig.PortID,
 		endpoint.ChannelID,
 		upgrade.Fields,
+		endpoint.Chain.GetSimApp().IBCKeeper.GetAuthority(),
+	)
+
+	proposal, err := govtypesv1.NewMsgSubmitProposal(
+		[]sdk.Msg{msg},
+		sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, govtypesv1.DefaultMinDepositTokens)),
 		endpoint.Chain.SenderAccount.GetAddress().String(),
+		endpoint.ChannelID,
+		"upgrade-init",
+		fmt.Sprintf("gov proposal for initialising channel upgrade: %s", endpoint.ChannelID),
+		false,
 	)
+	require.NoError(endpoint.Chain.TB, err)
 
-	return endpoint.Chain.sendMsgs(msg)
+	var proposalID int
+	res, err := endpoint.Chain.SendMsgs(proposal)
+	if err != nil {
+		return err
+	}
+
+	events := res.Events
+	for _, event := range events {
+		for _, attribute := range event.Attributes {
+			if attribute.Key == "proposal_id" {
+				proposalID, err = strconv.Atoi(attribute.Value)
+				require.NoError(endpoint.Chain.TB, err)
+			}
+		}
+	}
+
+	// vote on proposal
+	ctx := endpoint.Chain.GetContext()
+	require.NoError(endpoint.Chain.TB, endpoint.Chain.GetSimApp().GovKeeper.AddVote(ctx, uint64(proposalID), endpoint.Chain.SenderAccount.GetAddress(), govtypesv1.NewNonSplitVoteOption(govtypesv1.OptionYes), ""))
+	require.NoError(endpoint.Chain.TB, endpoint.Chain.GetSimApp().GovKeeper.AddVote(ctx, uint64(proposalID), endpoint.Chain.SenderAccount.GetAddress(), govtypesv1.NewNonSplitVoteOption(govtypesv1.OptionYes), ""))
+
+	// fast forward the chain context to end the voting period
+	params, err := endpoint.Chain.GetSimApp().GovKeeper.Params.Get(ctx)
+	require.NoError(endpoint.Chain.TB, err)
+
+	newHeader := endpoint.Chain.GetContext().BlockHeader()
+	newHeader.Time = endpoint.Chain.GetContext().BlockHeader().Time.Add(*params.MaxDepositPeriod).Add(*params.VotingPeriod)
+	ctx = ctx.WithBlockHeader(newHeader)
+	require.NoError(endpoint.Chain.TB, govtypes.EndBlocker(ctx, &endpoint.Chain.GetSimApp().GovKeeper))
```

Comment:
use [this](https://github.com/cosmos/ibc-go/blob/main/testing/coordinator.go#L57) function, you aren't incrementing the global time, so after executing this function, the chain is still in a state of pre proposal processing, it may lead to confusing side affects, you should be able to increment a block after incrementing the time and the proposal should be processed